pithos/pithos.py

```diff
@@ -22,13 +22,24 @@
 import signal
 
 import gi
-gi.require_version('Gst', '1.0')
 from gi.repository import Gst, GstPbutils, GObject, Gtk, Gdk, Pango, GdkPixbuf, Gio, GLib
 import contextlib
 import html
 import math
 import urllib.request, urllib.error, urllib.parse
 import json
+
+try:
+    gi.require_version('Gst', '1.0')
+except ValueError, e:
+    if e[0].startswith("Namespace Gst not available for version"):
+        print "Received error %s from Gstreamer." % e[0]
```

Comment:
I git cloned and typed python setup.py build. This of course ignores the 
hashbang line. The app itself should check for missing libs at runtime, 
since the system you build with is often not the same system it's 
deployed on. However, not trying to start an argument - I'm just 
naturally inclusive of even relatively old versions of *, and we may 
just have different philosophies of what's valuable or worth putting in 
extra time for. No worries :)


On 03/13/2015 12:07 PM, TingPing wrote:
>
> In pithos/pithos.py 
> <https://github.com/pithos/pithos/pull/158#discussion_r26403634>:
>
> >  from gi.repository import Gst, GstPbutils, GObject, Gtk, Gdk, Pango, GdkPixbuf, Gio, GLib
> >  import contextlib
> >  import html
> >  import math
> >  import urllib.request, urllib.error, urllib.parse
> >  import json
> > +
> > +try:
> > +    gi.require_version('Gst', '1.0')
> > +except ValueError, e:
> > +    if e[0].startswith("Namespace Gst not available for version"):
> > +        print "Received error %s from Gstreamer." % e[0]
>
> Except it does use python3 by default and you had to call it with 
> python2 manually... 
> https://github.com/pithos/pithos/blob/master/setup.py#L1
>
> Also setuptools is absolutely useless for checking for deps and since 
> they are runtime deps it would be counter productive for packagers to 
> force deps anyway.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/pithos/pithos/pull/158/files#r26403634>.
>